assests/javascript/index.js

```diff
@@ -1,46 +1,61 @@
+import cityListJson from "../json/cityList.json" assert { type: "json" };
+
 const apiKey = "8cf248a4fd398e2908cf52a9f187375b";
 const baseUrl = "https://api.openweathermap.org/";
 
 const mainSection = document.getElementById("main");
-const cityNameSelection = document.getElementById("cities");
+const selectedCity = document.getElementById("selected-city");
+const cityNameList = document.getElementById("cities");
 
-cityNameSelection.addEventListener("click", () => {
-  getWeather(cityNameSelection.value);
+selectedCity.addEventListener("keypress", (e) => {
+  if (e.key === "Enter") {
+    getWeather(getCityId(selectedCity.value));
+  }
 });
 
-getWeather(cityNameSelection.value);
+showDataList();
+getWeather(118743);
+
+function showDataList() {
+  return (cityNameList.innerHTML = cityListJson.map((city) => {
+      return `<option>${city.name}</option>`;
+    })
+    .join(""));
+}
 
-async function getWeather(cityName) {
+async function getWeather(cityId) {
   try {
-    const city = await getCityLocation(cityName);
-    const currentWeather = await getCurrentWeather(city);
-    const dailyForecastingWeatherList = await getForecastingWeather(city);
+    const currentWeather = await getCurrentWeather(cityId);
+    const dailyForecastingWeatherList = await getForecastingWeather(cityId);
 
     showWeather(currentWeather, dailyForecastingWeatherList);
   } catch {
     showErrorMassage();
   }
 }
 
-async function getCityLocation(cityName) {
-  const fetchedData = await fetch(
-    `${baseUrl}geo/1.0/direct?q=${cityName}&appid=${apiKey}`
-  );
-  const cityList = await fetchedData.json();
+function getCityId(cityName) {
+  let cityId = 0;
+
+  cityListJson.filter((city) => {
+    if (city.name === cityName) {
+      cityId = city.id;
+    }
+  });
 
-  return cityList[0];
+  return cityId;
 }
```

Comment:
Maybe?
```js
  const foundCity = cityListJson.find((city) => city.name === cityName);
  
  if (foundCity) {
    return foundCity.id
  }
  
  return 0;
```

Reply:
I didn't know there is **find** method.Thanks alllllotttttt ^^